packages/samples/headless-react/src/components/cart-recommendations-list/cart-recommendations-list.class.tsx

```diff
@@ -0,0 +1,135 @@
+import {Component} from 'react';
+import {
+  buildCartRecommendationsList,
+  CartRecommendationsList as HeadlessCartRecommendationsList,
+  CartRecommendationsListState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {productRecommendationEngine} from '../../engine';
+
+export interface CartRecommendationsListProps {
+  skus: string[];
+  maxNumberOfRecommendations?: number;
+}
+
+export class CartRecommendationsList extends Component<
+  CartRecommendationsListProps
+> {
+  private controller: HeadlessCartRecommendationsList;
+  public state: CartRecommendationsListState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: CartRecommendationsListProps) {
+    super(props);
+
+    this.controller = buildCartRecommendationsList(
+      productRecommendationEngine,
+      {
+        options: {
+          skus: this.props.skus,
+          maxNumberOfRecommendations: this.props.maxNumberOfRecommendations,
+        },
+      }
+    );
+    this.state = this.controller.state;
+  }
+
+  componentDidMount() {
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  render() {
+    if (this.state.error) {
+      return (
+        <div>
+          <div>Oops {this.state.error.message}</div>
+          <code>{JSON.stringify(this.state.error)}</code>
+          <button onClick={() => this.controller.refresh()}>Try again</button>
+        </div>
+      );
+    }
+
+    if (!this.state.recommendations.length) {
+      return <button onClick={() => this.controller.refresh()}>Refresh</button>;
+    }
+
+    return (
+      <div>
+        <button onClick={() => this.controller.refresh()}>Refresh</button>
+        Based on your cart, you may like the following:
+        <ul style={{textAlign: 'left'}}>
+          {this.state.recommendations.map((product) => (
+            <li key={product.sku}>
+              <article>
+                <h2>
+                  <a href={product.link}>{product.name}</a>
+                </h2>
+                <img
+                  src={product.thumbnailUrl}
+                  alt={`Thumbnail of ${product.name}`}
+                  style={{opacity: product.inStock ? 1 : 0.5}}
+                />
+                {product.promoPrice !== product.price ? (
```

Comment:
How exactly do we know if there's a promo price? Is it when it's different from the price, when it's -1, when it's NaN, or something else?

Reply:
I think when the field is present would be good enough

Reply:
Hmm, the field is marked as always present in the interface. PMed you.